src/Hooks/useSignUpMutation.js

```diff
@@ -0,0 +1,30 @@
+import SIGN_UP from "./../Apollo/Mutations/SignUp";
+import { useMutation } from "@apollo/react-hooks";
+
+const useSignUpMutation = () => {
+  const [mutate, { error, loading }] = useMutation(SIGN_UP, {
+    update: (cache, mutationResult) => {
+      localStorage.setItem("token", mutationResult.data.signUp.jwt);
+      cache.writeData({ data: { token: mutationResult.data.signUp.jwt } });
+    },
+  });
+  return {
+    error,
+    loading,
+    signUpUser: async (firstName, lastName, userName, passWord) => {
+      const { data } = await mutate({
+        variables: {
+          data: {
+            firstname: firstName,
+            lastname: lastName,
+            username: userName,
+            password: passWord,
+          },
+        },
+      });
+      return data && data.signUp;
+    },
+  };
+};
```

Comment:
To write less code you could just pass an object as parameter:

```suggestion
    signUpUser: async (input) => {
      const { data } = await mutate({
        variables: { data: input }
      });
      
      return data && data.signUp;
    },
  };
};
```
Then outside passing as parameter and object with the exact data structure that you need.